### PR TITLE
Feature/cvsb 10166 HGV Tech Record View detailed

### PR DESCRIPTION
**Pull request by m-ujaffer**
https://jira.dvsacloud.uk/browse/CVSB-10166

### CODE REVIEW

File: src/app/utils/testing.utils.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  function mockMicrofilm(args?: Partial<Microfilm>): Microfilm {
+    const mock: Microfilm = {
+      microfilmDocumentType: 'PSV Miscellaneous',
+      microfilmRollNumber: '1234',
+      microfilmSerialNumber: '5678'
+    };
+    return { ...mock, ...args };
+  }
+
+  function mockPlates(args?: Partial<Plate[]>): Plate[] {
```
Comment (alleypa) [inline src/app/utils/testing.utils.ts:46]: So this should be one object at a time. 

function mockPlates(args?: Partial<Plate>): Plate[] {
    const mock: Plate= {
        plateSerialNumber: '123123',
        plateIssueDate: '2019-12-13',
        plateReasonForIssue: '5678',
        plateIssuer: 'Issuer1'
    };
    return { ...mock, ...args };
  }

File: src/app/technical-record/plates/plates.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(PlatesComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should create view only with populated data', () => {
+    component.plates = [{
```
Comment (alleypa) [inline src/app/technical-record/plates/plates.component.spec.ts:26]: we can then use component.plates = [TESTING_UTILS.mockPlate()]. 
If you want variant of the object you can use component.plates = [
TESTING_UTILS.mockPlate(),
TESTING_UTILS.mockPlate({plateSerialNumber: 'Olu123')
]

File: src/app/technical-record/dimensions/dimensions.component.spec.ts [hunk 3]
```diff
@@ -1,73 +1,57 @@
...
+                              } as TechRecord;
+    component.activeRecord = tecRec;
+    component.axleSpacing = [
+                              { axles: '1', value: 100},
+                              { axles: '2', value: 100},
+                              { axles: '3', value: 100},
+                            ];
+  });
 
+  fit('should create view only with populated data', () => {
```
Comment (rspasici) [inline src/app/technical-record/dimensions/dimensions.component.spec.ts:49]: the f can be removed

File: src/app/technical-record/documents/documents.component.html [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    <div class="govuk-summary-list__row">
+      <dt class="govuk-summary-list__key">Microfilm serial number</dt>
+      <dd id="test-microfilm-microfilmSerialNumber" class="govuk-summary-list__value">
+        {{ microfilm?.microfilmSerialNumber | DefaultNullOrEmpty }}
+      </dd>
+      <dd class="govuk-summary-list__value"></dd>
+    </div>
+    </dl>
+</ng-container>
```
Comment (rspasici) [inline src/app/technical-record/documents/documents.component.html:27]: no eof line

File: src/app/technical-record/documents/documents.component.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    </div>
+
+    <div class="govuk-summary-list__row">
+      <dt class="govuk-summary-list__key">Microfilm serial number</dt>
+      <dd id="test-microfilm-microfilmSerialNumber" class="govuk-summary-list__value">
+        {{ microfilm?.microfilmSerialNumber | DefaultNullOrEmpty }}
+      </dd>
+      <dd class="govuk-summary-list__value"></dd>
+    </div>
+    </dl>
```
Comment (rspasici) [inline src/app/technical-record/documents/documents.component.html:26]: indent is off here or a tag is not properly closed

File: src/app/technical-record/dimensions/dimensions.component.html [hunk 6]
```diff
@@ -1,102 +1,128 @@
...
-    </td>
-    <th class="govuk-table__cell"></th>
-    <th class="govuk-table__cell"></th>
-    <th class="govuk-table__cell"></th>
-    <th class="govuk-table__cell"></th>
-  </tr>
+    <tr class="govuk-table__row">
+      <th class="govuk-table__cell govuk-!-font-weight-bold">Length (mm)</th>
+      <td id="test-length" class="govuk-table__cell">
+        {{ activeRecord?.dimensions.length | DefaultNullOrEmpty }}
```
Comment (alleypa) [inline src/app/technical-record/dimensions/dimensions.component.html:6]: better to have as this activeRecord?.dimensions?.length. length will cause the page to crash if not available

File: src/app/technical-record/dimensions/dimensions.component.html [hunk 7]
```diff
@@ -1,102 +1,128 @@
...
+      </td>
+      <th class="govuk-table__cell"></th>
```
Comment (alleypa) [inline src/app/technical-record/dimensions/dimensions.component.html:8]: Alot of <th class="govuk-table__cell"></th> in this file. I think we should remove all if not used

File: src/app/technical-record/dimensions/dimensions.component.html (hunk 6 above)
Comment (m-ujaffer) [reply src/app/technical-record/dimensions/dimensions.component.html:6]: fixed

File: src/app/technical-record/dimensions/dimensions.component.html (hunk 7 above)
Comment (m-ujaffer) [reply src/app/technical-record/dimensions/dimensions.component.html:8]: The complete component is designed as table. It needs the th to main the format

File: src/app/technical-record/technical-record.component.html [hunk 8]
```diff
@@ -383,12 +398,60 @@ <h1 class="govuk-heading-xl title">Technical record</h1>
             </mat-expansion-panel>
           </div>
 
+          <div class="auto-pad" [@.disabled]="true">
+            <mat-expansion-panel class="custom-header mat-elevation-z0 " hideToggle="true" #applicant
```
Comment (JonLazariniDD) [inline src/app/technical-record/technical-record.component.html:402]: As agreed, @irina005  had the changed implemented and waiting for ticket to refactor when possible.

reference: https://jira.dvsacloud.uk/browse/CVSB-15024

**Comment by m-ujaffer** [comment]
Closing in favour of common

**Pull request closed without merging**